### PR TITLE
feat: implement resizable-mixin

### LINKED ISSUE #27: Implement resize-observer-mixin
We should replace `IronResizableBehavior` with the native [Resize Observer](https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API).
This API is not supported by older Safari versions, so we need to load a [polyfill](https://github.com/PolymerLabs/uni-virtualizer/blob/e6d6b720e6027f73976c81a81b3b8e64a5beb616/packages/uni-virtualizer/src/lib/polyfillLoaders/ResizeObserver.ts).

The mixin should create an observer instance in the `connectedCallback` in order to observe element size change, and then call `disconnect()` inside of `disconnectedCallback`.

This mixin should be used in `vaadin-tabs`, `vaadin-form-layout` etc.

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #27

### CODE REVIEW

File: packages/resizable-mixin/resizable-mixin.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { ResizableClass } from './resizable-class';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+export const ResizableMixin = <T extends Constructor<LitElement>>(
+  base: T
+): T & Constructor<ResizableClass> => {
+  class Resizable extends base {
+    private _resizeObserver?: ResizeObserver;
```
Comment (tomivirkki) [inline packages/resizable-mixin/resizable-mixin.ts]: Would it make sense to only have 1 (static) instance of the ResizeObserver?
Something like:

```js
new ResizeObserver(entries) => {
  entries.forEach(entry => entry.target._sizeChanged(entry.contentRect));
});
```

Comment (web-padawan) [reply packages/resizable-mixin/resizable-mixin.ts]: Thanks, done.

File: packages/resizable-mixin/resizable-mixin.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  class Resizable extends base {
+    protected static _resizeObserver?: ResizeObserver;
+
+    connectedCallback() {
+      super.connectedCallback();
+
+      resizables.add(this);
+
+      this._initResizeObserver().then(() => {
+        const observer = Resizable._resizeObserver;
```
Comment (tomivirkki) [inline packages/resizable-mixin/resizable-mixin.ts]: Since this block is run async, should we check that the element is still connected?

File: packages/resizable-mixin/resizable-mixin.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        observer && observer.observe(this);
+        resizables.add(this);
```
Comment (tomivirkki) [inline packages/resizable-mixin/resizable-mixin.ts]: Is this also needed while this was already added to `resizables` synchronously in connectedCallback?

File: packages/resizable-mixin/resizable-mixin.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { LitElement } from 'lit-element';
+import getResizeObserver from './resize-observer-polyfill';
+import { ResizableClass } from './resizable-class';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+const resizables = new Set();
```
Comment (tomivirkki) [inline packages/resizable-mixin/resizable-mixin.ts]: Is this double bookkeeping necessary? I see it's used in disconnectedCallback to make sure all elements are unobserved when the set is empty. Shouldn't observe/unobserve logic in connected/disconnected callbacks be enough to make sure there are no unnecessary registrations?

File: packages/resizable-mixin/resizable-mixin.ts (hunk 2 above)
Comment (web-padawan) [reply packages/resizable-mixin/resizable-mixin.ts]: Good catch. Fixed, added a test to check for this edge case.

File: packages/resizable-mixin/resizable-mixin.ts (hunk 4 above)
Comment (web-padawan) [reply packages/resizable-mixin/resizable-mixin.ts]: Removed. I agree that it's not really necessary and just adds extra logic.

File: packages/resizable-mixin/resizable-mixin.ts (hunk 3 above)
Comment (web-padawan) [reply packages/resizable-mixin/resizable-mixin.ts]: Removed.

File: packages/resizable-mixin/resizable-mixin.ts (hunk 2 above)
Comment (tomivirkki) [reply packages/resizable-mixin/resizable-mixin.ts]: nit: there's an edge case for the edge case :) If the component is wrapped inside another wrapper element. Then the wrapper can get disconnected while `this.parentNode` still returns truthy.

Comment (web-padawan) [reply packages/resizable-mixin/resizable-mixin.ts]: Thanks, changed to use `isConnected` and added a test to cover that.
